bad-cert.js

```diff
@@ -0,0 +1,23 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket1 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
+        const good = '77:53:28:AD:42:B1:04:F7:49:2B:C7:C7:7B:2A:84:64:EA:0B:1F:CE';
+        const bad = 'mismatch';
+        console.log('Woohoo, www.google.com is using our pinned fingerprint');
+        return undefined;
+    }}, () => {
+    console.log('client connected');
+});
+
+socket1.on('error', (data)=> {
+    console.log(data);
+});
+
+var req = https.request({port: 1336, host: 'https://example2.com', rejectUnauthorized: false}, function(){ 
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- fa58e304d6cf569756343dc3b2e25a53 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Improper Certificate Validation
_Medium CWE-295_
The application disables certificate validation which leads to a rogue certificate not being rejected when its signature cannot be verified by any of the certificate authorities. This allows for an insecure connection and a man-in-the-middle attack.



### How to fix
Potential mitigations include the following:
  1. Certificates should be carefully managed and checked to assure that data are encrypted with the intended owner's public key.
  2. If certificate pinning is being used, ensure that all relevant properties of the certificate are fully validated before the certificate is pinned, including the hostname.